sqlx-core/src/connection.rs

```diff
@@ -49,6 +50,16 @@ pub trait Connection: Send {
     where
         Self: Sized;
 
+    /// Begin a new transaction with a custom statement.
+    ///
+    /// Returns a [`Transaction`] for controlling and tracking the new transaction.
+    fn begin_with(
+        &mut self,
+        statement: impl Into<Cow<'static, str>>,
+    ) -> BoxFuture<'_, Result<Transaction<'_, Self::Database>, Error>>
+    where
+        Self: Sized;
```

Comment:
Does this work as a default method that invokes `Transaction::begin()`? Then it's technically not a breaking change.

Reply:
Technically yes, but I think that might be awkward if someone expected the default to respect `statement`.

Reply:
That's fine though? It should just pass `statement` onto `TransactionManager::begin()` which has to change anyway. The difference with that trait is that it's semver-exempt.

Reply:
Oh my bad, I misread your original comment, thinking you had said to call `Connection::begin`. Yes, we should be able to call `Transaction::begin`. Is there a reason `Connection::begin` doesn't have a similar default impl?